enhancements/machine-config/auth-token.md

```diff
@@ -0,0 +1,139 @@
+---
+title: mco-auth-token
+authors:
+  - "@cgwalters"
+reviewers:
+  - "@crawford"
+approvers:
+  - "@ashcrow"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+creation-date: 2020-08-05
+last-updated: 2020-08-19
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Support a provisioning token for the Machine Config Server
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes that for new (e.g. 4.7+) installations a "provisioning token" is required by default
+to access the Machine Config Server.
+
+For IPI/machineAPI scenarios, this will be handled fully automatically.
+
+For user-provisioned installations, openshift-install will generate a token internally via the equivalent of
+
+```
+$ dd if=/dev/urandom bs=32 count=1 | base64 > token
+$ oc -n openshift-config create secret generic machineconfig-auth --from-file=token
+```
+
+And then this token will be injected into the "pointer ignition configs" it creates as a URL parameter, or
+potentially `Authorization: bearer`.
+
+
+## Motivation
+
+See https://github.com/openshift/machine-config-operator/pull/784
+
+The Ignition configuration can contain secrets, and we want to avoid it being accessible both inside and outside the cluster.
```

Comment:
Our Ignition config has contained the pull secret and the kubeconfig from the start in 4.1.  There's no course change here.
I've reworked the section to clarify that.

Reply:
There _is_ a change in course though. As I said, the intention from the start has been to remove secrets from the configs. This is a reversal from that goal and I would like to see it justified since it is pretty foundational to the design.